staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -600,6 +686,37 @@ func (p *sharedProcessor) addListenerLocked(listener *processorListener) {
 	p.syncingListeners = append(p.syncingListeners, listener)
 }
 
+func (p *sharedProcessor) removeListener(listener *processorListener) {
+	p.listenersLock.Lock()
+	defer p.listenersLock.Unlock()
+
+	if p.removeListenerLocked(listener) {
+		if p.listenersStarted {
+			close(listener.addCh)
+		}
+	}
+}
+
+func (p *sharedProcessor) removeListenerLocked(listener *processorListener) bool {
```

Comment:
Since this is only called one place, I'm not sure it really helps readability to split it out into a separate function like this, I personally would not.